parameter_sync.py

```diff
@@ -79,21 +84,22 @@ def get_parameters_with_prefix(prefix, next_token=None):
             query_result = ssm.describe_parameters(Filters=[{'Key': 'Name', 'Values': [prefix]}], NextToken=next_token)
         else:
             query_result = ssm.describe_parameters(Filters=[{'Key': 'Name', 'Values': [prefix]}])
-        logging.debug("Query result %s" % str(query_result))
+        logger.debug("Query result %s" % str(query_result))
         if 'ResponseMetadata' in query_result:
             if 'HTTPStatusCode' in query_result['ResponseMetadata']:
                 if query_result['ResponseMetadata']['HTTPStatusCode'] == 200:
                     if next_token is None:
                         #grab the parameter list on the first run or you'll lose it
                         parameter_list.extend(query_result['Parameters'])
                     if 'NextToken' in query_result:
-                        logging.debug("Next token found")
+                        logger.debug("Next token found")
                         parameter_list.extend(get_parameters_with_prefix(prefix, next_token=query_result['NextToken']))
-                        logging.debug("Out of recursion")
+                        parameter_list.extend(query_result['Parameters'])
```

Comment:
This would be the small change to make things work

Reply:
interesting the condition for 'if next_token is None' and 'no text token, storing' is two separate sections with the same code. I guess if it works, it works.

Reply:
This is a good point - just above the first if next_token you could just always extend the parameter list with the results, and then if there is a next token, then call the function again.

Reply:
the next_token is refering to an actual value at the time and the other one is refering to the value returned from AWS.  During the first run interration next_token will be None so that code will execute.  When you get the highest recursion next_token is set but the NextToken is no longer present in the query result.  So it is testing for 2 different things which are states at the bottom and top of the recursion.

Reply:
ah, so a local variable not related to the token from was. very interesting lol

Reply:
This is a bit of a mess, but I think in all cases, you want to extend the parameter list with the result of the query, but only in the case where NextToken is present do you want to recursively run the function again... You can leave it as is if you want, but it could be simplified I think.